apps/internal/base/internal/storage/storage.go

```diff
@@ -277,11 +272,10 @@ func (m *Manager) readAccessToken(homeID string, envAliases []string, realm, cli
 }
 
 func (m *Manager) writeAccessToken(accessToken AccessToken) error {
+	m.contractMu.Lock()
+	defer m.contractMu.Unlock()
 	key := accessToken.Key()
-
-	cache := m.Contract().copy()
-	cache.AccessTokens[key] = accessToken
-	m.contract.Store(cache)
+	m.contract.AccessTokens[key] = accessToken
```

Comment:
As a note, not for this line....

Did we run against this with the race detector?

Reply:
Yes, no problems were detected.